consensus/hotstuff/model/block.go

```diff
@@ -44,3 +44,12 @@ func GenesisBlockFromFlow(header *flow.Header) *Block {
 	}
 	return genesis
 }
+
+// CertifiedBlock holds a certified block, which is a block and a QC that pointing to it.
+// A QC is the aggregated form of votes from a supermajority of HotStuff participants.
+// Existence of a QC proves validity of the block. A certified block satisfies:
+// Block.View == QC.View and Block.BlockID == QC.BlockID
+type CertifiedBlock struct {
```

Comment:
Good idea to have it here, can we also not forget to cleanup follower and get rid of [`pending_tree.CertifiedBlock`](https://github.com/onflow/flow-go/blob/master/engine/common/follower/pending_tree/pending_tree.go#L14)?

Reply:
I think that is a good goal. I just tried it and got stuck: 
* `PendingTree`'s `CertifiedBlock` wraps `flow.Block`: https://github.com/onflow/flow-go/blob/b8f54ac1683b7ebb0ac373e1f3c05f7654cc8a4d/engine/common/follower/pending_tree/pending_tree.go#L14-L17
* While HotStuff's `CertifiedBlock` wraps [HotStuff's abstraction of a block](https://github.com/onflow/flow-go/blob/b8f54ac1683b7ebb0ac373e1f3c05f7654cc8a4d/consensus/hotstuff/model/block.go#L9-L18) https://github.com/onflow/flow-go/blob/b8f54ac1683b7ebb0ac373e1f3c05f7654cc8a4d/consensus/hotstuff/model/block.go#L52-L55

_Suggestion for further steps:_
* In the long term, I would like HotStuff to use an `Block` _interface_, which I think would remove a lot of those complications. Unfortunately, that is beyond the scope of this PR. I have created this issue: https://github.com/onflow/flow-go/issues/4170
* Within this PR, I have moved the `CertifiedBlock` to the `flow` package next to [`block`](https://github.com/onflow/flow-go/blob/c271a58af9f0e6f54fae061e24923a293fe0f7cc/model/flow/block.go#L29-L34)

Reply:
We could also think about using generics. CertifiedBlock could be both with `model.Block` and `flow.Block`